### PR TITLE
Fix various meaty comestibles' vitamin values

### LINKED ISSUE #48828: cooked meat has 10 times too much B12
# Describe the bug
1 piece of cooked meat has 778%  this is expected to be 77.8% or so, based on rough estimations from USDA.  2.4 micrograms needed per day, 100grams meat has ~.6 micrograms

# Steps To Reproduce

# Expected behavior

# Screenshots

# Versions and configuration

# Additional context

### PR DESCRIPTION
**Pull request by merrygin**
#### Summary
Bugfixes "Fixes various meat based food items' vitamin values"

#### Purpose of change
Fixes #48828. Some meat based comestibles had inflated and/or wrong and/or missing vitamin values. Due to many inheritances from meat_cooked, this concerns quite a few items (mutant variety, too).

#### Describe the solution
I adapted various vitamin values, adding some when they were missing. I used USDA informational websites to generate respective values (using pork as default meat, if data was available). 
I used [https://reedir.arsnet.usda.gov/codesearchwebapp/(S(qqrqzk3frf2prudap0utjwi1))/codesearch.aspx](url) (USDA's "What's in the Foods You Eat"-tool; clickign the link doesn't work for some reason, but copy-pasting the link works fine) to search for food items nutritional values. I used [https://nutrition.org/nutrient-information/](url) (American Society for Nutrition) for Recommended Daily Average values to determine in-game RDA.
Vitamin values should now be more complete and more accurate for many meaty comestibles.

#### Describe alternatives you've considered
Many of the meaty comestibles probably shouldn't inherit vitamins from meat_cooked. Also, there probably should be different items/values for different types of meat (pork, beef, etc.) since they have very different nutritional values. Because this would be a whole other issue, I used pork as default.

#### Testing
Generated some of the items in-game and checked if new values applied. They did.

#### Additional context
This is my first PR - I am very sorry for any newbie mistakes! 👶

### CODE REVIEW

**Comment by merrygin** [comment]
As an addendum, offal values are now pretty insane, since they apparently are in reallife. Its debateable if this could be a balance issue.

File: data/json/items/comestibles/carnivore.json [hunk 1]
```diff
@@ -226,7 +226,7 @@
     "cooks_like": "meat_scrap_cooked",
     "proportional": { "weight": 0.1, "volume": 0.1, "price": 0.1, "calories": 0.1 },
     "delete": { "flags": [ "SMOKABLE" ] },
-    "vitamins": [ [ "vitA", 0 ], [ "vitC", 0 ], [ "calcium", 0 ], [ "iron", 0 ], [ "vitB", 0 ] ]
+    "vitamins": [ [ "calcium", 0 ], [ "iron", 1 ], [ "vitB", 8 ] ]
```
Comment (BrettDong) [inline data/json/items/comestibles/carnivore.json:229]: Trailing white spaces:
```suggestion
    "vitamins": [ [ "calcium", 0 ], [ "iron", 1 ], [ "vitB", 8 ] ]
```

File: data/json/items/comestibles/meat_dishes.json [hunk 2]
```diff
@@ -149,7 +149,7 @@
     "material": [ "flesh" ],
     "charges": 2,
     "flags": [ "EATEN_HOT", "SMOKED" ],
-    "vitamins": [ [ "calcium", 0 ], [ "iron", 2 ] ],
+    "vitamins": [ [ "vitA", 1 ],[ "vitC", 0 ],[ "calcium", 0 ], [ "iron", 6 ], [ "vitB", 39 ] ],
```
Comment (BrettDong) [inline data/json/items/comestibles/meat_dishes.json:152]: Missing spaces:
```suggestion
    "vitamins": [ [ "vitA", 1 ], [ "vitC", 0 ], [ "calcium", 0 ], [ "iron", 6 ], [ "vitB", 39 ] ],
```

File: data/json/items/comestibles/meat_dishes.json [hunk 3]
```diff
@@ -196,6 +196,7 @@
     "material": [ "flesh" ],
     "charges": 4,
     "flags": [ "EATEN_HOT" ],
+    "vitamins": [ [ "vitA", 1 ],[ "vitC", 0 ],[ "calcium", 0 ], [ "iron", 3 ], [ "vitB", 19 ] ],
```
Comment (BrettDong) [inline data/json/items/comestibles/meat_dishes.json:199]:
```suggestion
    "vitamins": [ [ "vitA", 1 ], [ "vitC", 0 ], [ "calcium", 0 ], [ "iron", 3 ], [ "vitB", 19 ] ],
```

File: data/json/items/comestibles/meat_dishes.json [hunk 4]
```diff
@@ -444,14 +445,14 @@
...
-    "calories": 26,
+    "calories": 157,
     "description": "Formally known as Bologna sausage, this is a finely ground preserved meat that comes in circular slices.  Its first name isn't Oscar.  You can eat it cold.",
     "price": 1000,
     "price_postapoc": 250,
     "material": [ "flesh" ],
     "volume": "750 ml",
     "charges": 10,
-    "vitamins": [ [ "iron", 15 ], [ "vitB", 8 ] ]
+    "vitamins": [ [ "vitA", 0 ],[ "vitC", 0 ],[ "calcium", 6 ], [ "iron", 5 ], [ "vitB", 11 ] ]
```
Comment (BrettDong) [inline data/json/items/comestibles/meat_dishes.json:455]:
```suggestion
    "vitamins": [ [ "vitA", 0 ], [ "vitC", 0 ], [ "calcium", 6 ], [ "iron", 5 ], [ "vitB", 11 ] ]
```

File: data/json/items/comestibles/meat_dishes.json [hunk 5]
```diff
@@ -581,7 +582,7 @@
     "price_postapoc": 350,
     "material": [ "flesh", "veggy" ],
     "volume": "120 ml",
-    "vitamins": [ [ "vitC", 3 ], [ "calcium", 24 ], [ "iron", 14 ] ],
+    "vitamins": [ [ "vitA", 14 ],[ "vitC", 3 ], [ "calcium", 5 ], [ "iron", 21 ] ],
```
Comment (BrettDong) [inline data/json/items/comestibles/meat_dishes.json:585]:
```suggestion
    "vitamins": [ [ "vitA", 14 ], [ "vitC", 3 ], [ "calcium", 5 ], [ "iron", 21 ] ],
```

**Review by BrettDong — COMMENTED** [review]
There are some JSON formatting errors. You can use this online tool to format your edited JSON data before making the `git` commit: https://dev.narc.ro/cataclysm/format.html

File: data/json/items/comestibles/meat_dishes.json (hunk 5 above)
Comment (merrygin) [reply data/json/items/comestibles/meat_dishes.json:585]: Very sorry for the inconvenience BrettDong. I thought I would be able to do it without a tool or something, which was obviously a bad call. ^^' 
How should I proceed? Can I still update this PR through converting it into a draft, or should I cancel, make the changes and commit again?

Comment (Zireael07) [reply data/json/items/comestibles/meat_dishes.json:585]: You should be able to update this PR just by making a new commit. Unless there are many changes to be made, there shouldn't be any need to mark the PR as draft.

Comment (merrygin) [reply data/json/items/comestibles/meat_dishes.json:585]: Ok cool, thank you.

Comment (merrygin) [reply data/json/items/comestibles/meat_dishes.json]: Actually, i would have another question if you (or anybody else) don't mind. I made the changes by hand now (for the learning effect of doing it properly next time ;) ) but tried the linting tool, too. 
It gave me a very different output from the format used in the JSON-file. Is that a matter of the file not being updated to the recommended formatting, or an issue of the tool not giving the right output (or an issue of me, not realising something else :D ).

**Comment by actual-nh** [comment]
You need to take a look at the recipe for bologna & its results - calories at least - and compare that to the bologna food item. That's the problem with (at least the current) the Travis CI build.

**Comment by merrygin** [comment]
> 
> 
> You need to take a look at the recipe for bologna & its results - calories at least - and compare that to the bologna food item. That's the problem with (at least the current) the Travis CI build.

Oh ok, I will do that, thanks for the info. I actually think I should not have tinkered with the calories in this PR at all, since it is about vitamins... I may just revert that change altogether.

**Pull request merged**